types/layui-src/misc.d.ts

```diff
@@ -1,3546 +1,3542 @@
+
+        /**
+         * 控件初始打开的回调
+         * @param [date] 基础参数
+         */
+        ready?(dateParam: DateParam): void;
+
+        /**
+         * 日期时间被切换后的回调   this to test and elem
+         * @param [value] 得到日期生成的值，如：2017-08-18 范围："2021-07-06 - 2021-08-09"
+         * @param [date] 得到日期时间对象：{year: 2017, month: 8, date: 18, hours: 0, minutes: 0, seconds: 0}
+         * @param [endDate] 开启范围选择（range: true）才会返回。对象成员同上date
+         */
+        change?(this: Required<DateOption>, value: string, date: DateParam, endDate: DateParam): void;
+
+        /**
+         * 控件选择完毕后的回调
+         * @param [value] 得到日期生成的值，如：2017-08-18 范围："2021-07-06 - 2021-08-09"
+         * @param [date] 得到日期时间对象：{year: 2017, month: 8, date: 18, hours: 0, minutes: 0, seconds: 0}
+         * @param [endDate] 开启范围选择（range: true）才会返回。对象成员同上date
+         */
+        done?(value: string, date: DateParam, endDate: DateParam): void;
+    }
+
+    /**
+     * 日期和时间组件
+     */
+    interface Laydate {
+        /**
+         * 核心方法
+         * @param [options]  基础参数
+         */
+        render(options: DateOption): { config: DateOption; hint: (content: string) => void };
+
+        /**
+         * 设置全局参数
+         * @param [options]
+         */
+        set(options?: DateOption): Laydate;
+
+        /**
+         * 配置基础路径    <br/>&nbsp;
+         *  如果你不是采用 layui 或者普通 script 标签方式加载的 laydate.js，    <br/>&nbsp;
+         *  而是采用 requirejs 等其它方式引用 laydate，    <br/>&nbsp;
+         *  那么你需要设置基础路径，以便 laydate.css 完成加载。
+         */
+
+        path: string;
+
+        /**
+         * 获取指定年月的最后一天
+         * @param [month]  month默认为当前月
+         * @param [year]  year默认为当前年
+         */
+        getEndDate(month?: number, year?: number): number;
+    }
+
+    /**
+     * 编辑器基本设置
+     */
+    interface EditOption {
+        /**
+         * 重新定制编辑器工具栏，如： tool: ['link', 'unlink', 'face'] 当前可选有：    <br/>&nbsp;
+         *             'strong' //加粗    <br/>&nbsp;
+         *             ,'italic' //斜体    <br/>&nbsp;
+         *             ,'underline' //下划线    <br/>&nbsp;
+         *             ,'del' //删除线    <br/>&nbsp;
+         *
+         *             ,'|' //分割线    <br/>&nbsp;
+         *
+         *             ,'left' //左对齐    <br/>&nbsp;
+         *             ,'center' //居中对齐    <br/>&nbsp;
+         *             ,'right' //右对齐    <br/>&nbsp;
+         *             ,'link' //超链接    <br/>&nbsp;
+         *             ,'unlink' //清除链接    <br/>&nbsp;
+         *             ,'face' //表情    <br/>&nbsp;
+         *             ,'image' //插入图片    <br/>&nbsp;
+         *             ,'help' //帮助    <br/>&nbsp;
+         */
+
+        tool?: string[];
+        /**
+         * 不显示编辑器工具栏，一般用于隐藏默认配置的工具bar
+         */
+        hideTool?: string[];
+        /**
+         * 设定编辑器的初始高度
+         */
+        height?: number | string;
+        /**
+         * 设定图片上传接口，如：uploadImage: {url: '/upload/', type: 'post'},需要服务端返回：    <br/>&nbsp;
+         *   {	<br/>&nbsp;
```

Comment:
<!--
  1 failure:  The root .editorc...
  0 warning: 
  
  
  DangerID: danger-id-Danger;
  File: types/layui-src/misc.d.ts;
  Line: 1186;
-->
- :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.